lib/ddtrace/contrib/cucumber/formatter.rb

```diff
@@ -0,0 +1,104 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/ci'
+require 'ddtrace/ext/test'
+require 'ddtrace/contrib/analytics'
+require 'ddtrace/contrib/cucumber/ext'
+
+module Datadog
+  module Contrib
+    module Cucumber
+      # Defines collection of instrumented Cucumber events
+      class Formatter
+        attr_reader :config
+        private :config
+
+        attr_reader :current_feature_span, :current_step_span
+        private :current_feature_span, :current_step_span
+
+        def initialize(config)
+          @config = config
+
+          bind_events(config)
+        end
+
+        def bind_events(config)
+          config.on_event :test_case_started, &method(:on_test_case_started)
+          config.on_event :test_case_finished, &method(:on_test_case_finished)
+          config.on_event :test_step_started, &method(:on_test_step_started)
+          config.on_event :test_step_finished, &method(:on_test_step_finished)
+        end
+
+        def on_test_case_started(event)
+          trace_options = {
+            app: Ext::APP,
+            resource: event.test_case.name,
+            service: configuration[:service_name],
+            span_type: Datadog::Ext::AppTypes::TEST,
+            tags: tags.merge(Datadog.configuration.tags)
+          }
+          @current_feature_span = tracer.trace(Datadog::Ext::AppTypes::TEST, trace_options)
```

Comment:
The first argument to `tracer.trace` is the operation name. All our existing operations names have the form: `integration.operation`.

In this case, I would expect `cucumber.test` (or similar), as this is measuring a single Cucumber test run.

Having it as simply `test` would cause confusion when another testing framework is added.

Reply:
I am using metrics (+alerts) for tests across different frameworks. Would there be a way to have a single test trace metric generated if a repository is using different frameworks?

Reply:
In theory, the newly introduced `span_type: 'test'` would uniquely give you metrics for spans related to test runs. I'm not sure if there's a way in the backend to get metrics by span type though.

Reply:
I noticed that in Java we use `framework.operation` as the span name for CI instrumentation (`junit.test` in their case): https://github.com/DataDog/dd-trace-java/blob/ad15655605e6b6a8b040ef4405dd99ee891a6e37/dd-java-agent/instrumentation/junit-4.10/src/main/java/datadog/trace/instrumentation/junit4/TracingListener.java#L27

Maybe we can make it work with `cucumber.test`, the same way they did.

Reply:
I need to find a way to override this so it won't affect our test metrics.

Reply:
I have added configuration option `:operation_name` for `:cucumber` integration to solve this issue.